### ISSUE TITLE
Latest Native Plugin - cant build on Android

### ISSUE THREAD

**Issue by rotter** [googleads/googleads-mobile-unity#920]
I have downloaded the latest beta plugin for Native Ads today (unfortunately native plugins dont have version on their zip filename) and I am unable to build on Android.

I get the following errors:
GoogleMobileAds_2.cpp:32726: error undefined reference to 'GADURelease'
GoogleMobileAds_2.cpp:32728: error undefined reference to 'GADUCreateSmartBannerView'
GoogleMobileAds_2.cpp:32740: error undefined reference to 'GADUCreateBannerView'
GoogleMobileAds_2.cpp:32832: error: undefined reference to 'GADUSetBannerCallbacks'
(dozens of similar lines)

Any ideas? I will retry, and also will return to the previous version I had which was working (maybe 3.16.0.1?).

**Comment by stowy** [comment]
Hi,

This is odd because those symbols are only defined in the iOS plugin.

Can you try removing all GoogleMobileAds code and re-importing?

**Comment by rotter** [comment]
I have returned to the old plugin and it still doesnt build (with the same error), so something else is messed up. I will investigate a bit more and write back.

**Comment by rotter** [comment]
Ok, now it worked on the old version, still havent tested on the new version but it should work as well. I deleted the Library and Obj folders from my project to clear the cache/compiled files/etc.

This problem happened when I tried to upgrade the plugins, but first I forgot I was using the beta plugin, so I installed the regular one (which has the iOS GADU* files), then it gave me compile errors on the NativeAds part of my code (as expected), then I remembered I had to download the Beta 
plugin and downloaded it and installed, so I think something messed up and the binaries got cached and it would break the build. Now why those iOS files/code would break the Android build is beyond me.

**Comment by rotter** [comment]
Today when I tried to build, same problem happened. Then I deleted the same folders and it's building again (I have built 2-3 times already and it worked fine, I have even closed/reopened Unity and also building fine). Something is off somewhere. If I get any updates I will post.

**Comment by rotter** [comment]
Same thing happened today, can't build once again.

I think it's related to PlayServicesResolver, every time this bugs happens I notice on git that "Assets/Plugins/Android/com.google.android.gms.play-services-ads-lite-17.2.0.aar" is modified.

**Comment by rotter** [comment]
I checked the Editor Log (Console - Editor Log) to see the build log and noticed there were some temporary filed being used in /var/folders/.../

So I went there and deleted everything, tried to build again and it worked.

Will continue monitoring to see if it was an old/bugged file, or if it gets recreated from time to time and causes the problem again.

**Comment by rotter** [comment]
Same build problem happened later yesterday, and now again. I am deleting the Libraries and Obj folders from my project (instead of /var/folders/something/) and it works again. I will now completely stop using the new Admob plugin version and see the problem stops over the next days.

**Comment by stowy** [comment]
This issue doesn't make sense to me. Are you able to provide a reproducing
project? Can you screenshot the import screen when you add the plugin? I
don't fully grasp how it is that I haven't seen this issue in my local
testing.  It sounds to me like the iOS CS code is somehow getting compiled
into the Android code.  But this was not happening to me. Are you using
Mono or IL2CPP?

**Comment by rotter** [comment]
Thanks for the reply. I havent reopened the issue yet because I cant fully put the blame on the new version of the plugin, it could be a Unity bug or something else. The final test is the one Im doing now which is to revert the plugin to the old version(3.16.0.1 I think) and see if the problem fully stops.

Some specs:
- IL2CPP
- .NET 4.x
- Im on a Mac OSX
- PlayServiceResolver was .111 until yesterday, I then updated it to .112 and error persisted
- Latest Firebase (updated it a couple days ago), using Analytics module only

**Comment by stowy** [comment]
Ok IL2CPP could be the issue. I haven't verified the plugin with IL2CPP.

**Comment by rotter** [comment]
I have reverted the plugin to 3.16.0.1 and am still getting the same build problem, so it seems it's not related to the new plugin version.

One thing that occurred to me that I have recently enabled for this project is "App Bundle". Maybe this could be the culprit?

**Comment by AllanRW** [comment]
I'm also facing the same issues, I've have to restart Unity to fix the issue... 
I'm also using IL2CPP, App bundle, don't import Google.MiniJson.dll because of Firebase, Unity 2019.1.5f1

**Comment by rotter** [comment]
I havent updated this thread because Im focusing on other games which dont use Native Ads plugin so I dont have this problem on them... but I believe the plugin is indeed with problems. Can any dev confirm?

**Comment by stowy** [comment]
I believe this is IL2CPP specific, and a developer has created a pull request that they believe resolves the issue here - https://github.com/googleads/googleads-mobile-unity/pull/930.  Can you test and see if this helps?

**Comment by AllanRW** [comment]
I've tested with the #930 pull request but it doesn't fix the issue, same errors.

**Comment by PayandaOnline** [comment]
> 
> 
> I've tested with the #930 pull request but it doesn't fix the issue, same errors.

Go to Player Settings Select "Managed Stripping Level" to "Low" and Rebuild Your App.
 I'm waiting your reply.

**Comment by AllanRW** [comment]
I already have this setting, but today I've done more than 10 builds and didn't get the errors, but I haven't done any important changes on project, will keep testing through next week.

**Comment by AllanRW** [comment]
Just got the errors again after a few more builds, I've only made script changes on the project.

**Comment by Hardik-Nadiyapara** [comment]
i have same errors while building apk

**Comment by adnanmiczon** [comment]
i have downloaded google mobile ads Sdk for native ads from here (NATIVE ADS BUILD  Unity pakage-which support native ads)

https://developers.google.com/admob/unity/native-advanced-unified

Then i am integrating in Unity 2019.1.0f2 (64-bit). Setup everything it work fine on Mono but when i change to 64-bit build for final uploading
During building apk Unity Stucks on Building native binary with IL2CP...

After that Unity Gives this Error

Failed running C:\Program Files\Unity\Editor\Data\il2cpp/build/il2cpp.exe --convert-to-cpp --emit-null-checks --enable-array-bounds-check --dotnetprofile="unityaot" --compile-cpp --libil2cpp-static --platform="Android" --architecture="ARMv7" --configuration="Release" --outputpath="F:\RaceAwardWin\Temp\StagingArea\assets\bin\Data\Native\armeabi-v7a\libil2cpp.so" --cachedirectory="F:\RaceAwardWin\Assets\..\Library\il2cpp_android_armeabi-v7a/il2cpp_cache" --additional-include-directories="C:\Program Files\Unity\Editor\Data\PlaybackEngines\AndroidPlayer/Tools\bdwgc/include" --additional-include-directories="C:\Program Files\Unity\Editor\Data\PlaybackEngines\AndroidPlayer/Tools\libil2cpp/include" --tool-chain-path="F:/android-ndk-r16b-windows-x86_64/android-ndk-r16b" --profiler-report --map-file-parser="C:/Program Files/Unity/Editor/Data/Tools/MapFileParser/MapFileParser.exe" --directory=F:/RaceAwardWin/Temp/StagingArea/assets/bin/Data/Managed --generatedcppdir=F:/RaceAwardWin/Temp/StagingArea/Il2Cpp/il2cppOutput 

stdout:

Building libil2cpp.so with AndroidToolChain

	Output directory: F:\RaceAwardWin\Temp\StagingArea\assets\bin\Data\Native\armeabi-v7a

	Cache directory: F:\RaceAwardWin\Library\il2cpp_android_armeabi-v7a\il2cpp_cache

ObjectFiles: 235 of which compiled: 219

	Time Compile: 25560 milliseconds Il2CppAttributes.cpp

	Time Compile: 17582 milliseconds Il2CppCompilerCalculateTypeValues1.cpp

	Time Compile: 15864 milliseconds Il2CppTypeDefinitions.c

	Time Compile: 15672 milliseconds Il2CppMetadataUsage.c

	Time Compile: 14089 milliseconds Assembly-CSharp3.cpp

	Time Compile: 10597 milliseconds ProBuilderCore-Unity5.cpp

	Time Compile: 9720 milliseconds Il2CppInvokerTable.cpp

	Time Compile: 5484 milliseconds Generics18.cpp

	Time Compile: 5444 milliseconds ProBuilderCore-Unity51.cpp

	Time Compile: 5388 milliseconds mscorlib3.cpp

Total compilation time: 151626 milliseconds.

il2cpp.exe didn't catch exception: Unity.IL2CPP.Building.BuilderFailedException: F:\android-ndk-r16b-windows-x86_64\android-ndk-r16b\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++ @"C:\Users\Miczon LLP\AppData\Local\Temp\tmpEC90.tmp" -o "F:\RaceAwardWin\Library\il2cpp_android_armeabi-v7a\il2cpp_cache\linkresult_F69D8FD532EC526F74AA591119FF6C7F\libil2cpp.so" -shared -Wl,-soname,libil2cpp.so -Wl,--no-undefined -Wl,-z,noexecstack -Wl,--gc-sections -Wl,--build-id --sysroot "F:\android-ndk-r16b-windows-x86_64\android-ndk-r16b\platforms\android-16\arch-arm" -gcc-toolchain "F:\android-ndk-r16b-windows-x86_64\android-ndk-r16b\toolchains\arm-linux-androideabi-4.9\prebuilt\windows-x86_64" -target armv7-none-linux-androideabi -Wl,--wrap,sigaction -L "F:\android-ndk-r16b-windows-x86_64\android-ndk-r16b\sources\cxx-stl\gnu-libstdc++\4.9\libs\armeabi-v7a" -lgnustl_static -llog -rdynamic -fuse-ld=gold.exe



F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14500: error: undefined reference to 'GADURelease'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14500: error: undefined reference to 'GADURelease'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:15023: error: undefined reference to 'GADUCreateRewardedAd'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14500: error: undefined reference to 'GADURelease'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:15023: error: undefined reference to 'GADUCreateRewardedAd'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:15101: error: undefined reference to 'GADUSetRewardedAdCallbacks'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:15065: error: undefined reference to 'GADURequestRewardedAd'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14500: error: undefined reference to 'GADURelease'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14302: error: undefined reference to 'GADUCreateRequest'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14400: error: undefined reference to 'GADUAddKeyword'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14380: error: undefined reference to 'GADUAddTestDevice'

F:\RaceAwardWin\Temp\StagingArea\Il2Cpp\il2cppOutput/GoogleMobileAds2.cpp:14428: error: undefined


Unity Generates some files in Temp. GoogleMobileAds2 file. i dnt knw why it is generating this file on runtime

i dont have any IOS file in my project, i already tried with different versions of unity and try every thing which mention on different forums. but my problem does not solved.
please help in this matter

**Comment by rotter** [comment]
I tried adding Native Ads today to a game that didnt have Native yet, just downloaded the 4.1.0.1 Native Build and right on my 1st build attempt I got this exact same error. Luckily I remembered I had ran into this some time ago (6 months ago, it seems), so I didnt waste any time on this.

This other game also uses IL2CPP and App Bundle, and Im on Unity 2019.1.14f.

Any ideas/workarounds/fixes coming soon? Deleting the Library folder doesnt seem to be a safe way to fix this, so I dont feel comfortable deploying the game in production with that "fix", so I think I will just have to roll back the Native ad implementation.
